source/Energinet.DataHub.Charges.Libraries/source/Energinet.DataHub.Charges.Clients.Tests/Charges/ChargesClientTests.cs

```diff
@@ -143,35 +144,73 @@ public class ChargesClientTests
             var result = await sut.GetChargesAsync().ConfigureAwait(false);
 
             // Assert
-            result.Should().BeNull();
+            result.Should().BeEmpty();
         }
 
         [Theory]
         [InlineAutoDomainData]
-        public async Task SearchChargesAsync_WhenResponseIsNotFound_ReturnsEmptyList(
+        public async Task SearchChargesAsync_WhenResponseIsNotSuccess_ThrowsException(
             Mock<IChargesClientFactory> chargesClientFactory,
-            SearchCriteriaV1Dto searchCriteria)
+            ChargeSearchCriteriaV1Dto searchCriteria)
         {
             // Arrange
-            var mockHttpMessageHandler = GetMockHttpMessageHandler(HttpStatusCode.NotFound, string.Empty);
+            var mockHttpMessageHandler = GetMockHttpMessageHandler(HttpStatusCode.InternalServerError, string.Empty);
+            var httpClient = CreateHttpClient(mockHttpMessageHandler);
+            chargesClientFactory.Setup(x => x.CreateClient(httpClient))
+                .Returns(new ChargesClient(httpClient));
+
+            var sut = chargesClientFactory.Object.CreateClient(httpClient);
+
+            // Act / Assert
+            await Assert.ThrowsAsync<Exception>(async () => await sut.SearchChargesAsync(searchCriteria).ConfigureAwait(false)).ConfigureAwait(false);
+        }
+
+        [Theory]
+        [InlineAutoDomainData]
+        public async Task SearchChargesAsync_WhenResponseIsBadRequest_ThrowsExceptionWithMessage(
+            Mock<IChargesClientFactory> chargesClientFactory,
+            ChargeSearchCriteriaV1Dto searchCriteria)
+        {
+            // Arrange
+            var message = "Validation not valid";
+            var mockHttpMessageHandler = GetMockHttpMessageHandler(HttpStatusCode.BadRequest, message);
+            var httpClient = CreateHttpClient(mockHttpMessageHandler);
+            chargesClientFactory.Setup(x => x.CreateClient(httpClient))
+                .Returns(new ChargesClient(httpClient));
+
+            var sut = chargesClientFactory.Object.CreateClient(httpClient);
+
+            // Act / Assert
+            var ex = await Assert.ThrowsAsync<Exception>(async () => await sut.SearchChargesAsync(searchCriteria).ConfigureAwait(false)).ConfigureAwait(false);
+            ex.Message.Should().Contain(message);
```

Comment:
Test to also cover the exception message thrown:

```suggestion
            var responseBody = "Validation not valid";
            var exceptionMessage = $"Charges backend returned HTTP status code {(int)HttpStatusCode.BadRequest} with message {responseBody}";
            var mockHttpMessageHandler = GetMockHttpMessageHandler(HttpStatusCode.BadRequest, exceptionMessage);
            var httpClient = CreateHttpClient(mockHttpMessageHandler);
            chargesClientFactory.Setup(x => x.CreateClient(httpClient))
                .Returns(new ChargesClient(httpClient));

            var sut = chargesClientFactory.Object.CreateClient(httpClient);

            // Act / Assert
            var ex = await Assert.ThrowsAsync<Exception>(async () => await sut.SearchChargesAsync(searchCriteria).ConfigureAwait(false)).ConfigureAwait(false);
            ex.Message.Should().Be(exceptionMessage );
```